packages/zipkin/index.d.ts

```diff
@@ -239,7 +239,7 @@ declare namespace zipkin {
       constructor(key: string, value: boolean | string | number);
       readonly annotationType: string;
       key: string;
-      value: string;
+      value: boolean | string | number;
```

Comment:
I don't think we should change this, annotations should be only strings as that is what zipkin supports.

Reply:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

Reply:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

Reply:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

Reply:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

Reply:
I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript..

Reply:
I think the last one is enough (cc @adriancole), I am not sure if we break existing users by this (I guess not). Just one question: do we need `readonly annotationType: string;` in the last case?

Reply:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript..
The tests failed is my tests on type-tests folder, I saw the constructor accept boolean|string|number so I wrote tests to verify those three cases. But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the last case?
It is mandatory I guess since BinaryAnnotation implement the IAnnotation interface, I tried to remove it but getting errors on compiling. And also, the actual implementation has annotationType property.

Reply:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript.
The tests failed is my tests on type-tests folder, I saw the constructor accept boolean|string|number, so I wrote tests to verify those three cases. But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the last case?
It is mandatory I guess since BinaryAnnotation implements the IAnnotation interface, I tried to remove it but getting errors on compiling. And also, the actual implementation has annotationType property.

Reply:
@jcchavezs @adriancole I updated the PR so with BinaryAnnotation the constructor will only allow string value.

> I think we constrain the typescript api to string, if the underlying javascript can still pass otherwise. I didn't realize that the typescript will affect unit tests written in javascript.

The tests failed is my tests on type-tests folder, I saw the constructor accept boolean|string|number, so I wrote tests to verify those three cases. But now when we only accept string then I also modified the ts tests.

> Just one question: do we need `readonly annotationType: string;` in the last case?

It is mandatory I guess since BinaryAnnotation implements the IAnnotation interface, I tried to remove it but getting errors on compiling. And also, the actual implementation has annotationType property.